test/fixtures/dialects/mysql/alter_database.sql

```diff
@@ -0,0 +1,26 @@
+ALTER DATABASE my_database
+DEFAULT CHARACTER SET utf8mb4
+COLLATE utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION 'N';
+
+ALTER DATABASE my_database
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci
+DEFAULT ENCRYPTION = 'N';
+
+ALTER SCHEMA my_database
```

Comment:
Same here - also, not sure if intended, but you haven't changed the alter schema grammar at all

Reply:
On MySQL, ALTER SCHEMA is a synonym for ALTER DATABASE, as described in https://dev.mysql.com/doc/refman/8.0/en/alter-database.html.

So, I've implemented it in AlterDatabaseStatementSegment.